### PR TITLE
Added param checks for method parse_search_string().

### PR DESCRIPTION
**Pull request by manwar**
Hi @oalders 

Please review the PR.
Added basic param validation to the method parse_search_string().

Many Thanks.
Best Regards,
Mohammad S Anwar

### CODE REVIEW

File: t/005_parse_more.t [hunk 1]
```diff
@@ -39,6 +39,9 @@ if ( exists $ENV{'TEST_UPM_CACHED'}
     diag('caching is enabled...');
 }
 
+eval { $more->parse_search_string };
+like ($@, qr/missing url/);
```
Comment (oalders) [inline t/005_parse_more.t]: How about using `Test::Fatal` here?

```perl
use Test::Fatal;

...

like(
  exception { $more->parse_search_string;  },
  qr/missing url/,
  'dies on missing URL',
);

**Review by oalders — CHANGES_REQUESTED** [review]

**Comment by oalders** [comment]
Thanks @manwar! Just added a comment for you to consider.

**Comment by manwar** [comment]
Done as suggested.
Please review it again.

Best Regards,
Mohammad S Anwar

**Review by oalders — APPROVED** [review]

**Comment by oalders** [comment]
Wonderful. Thanks @manwar!

**Pull request merged**